src/main/java/bio/terra/service/filedata/azure/AzureSynapsePdao.java

```diff
@@ -81,7 +85,10 @@
           + "    FROM OPENROWSET(\n"
           + "       BULK '<ingestFileName>',\n"
           + "       DATA_SOURCE = '<ingestFileDataSourceName>',\n"
-          + "       FORMAT = 'parquet') AS rows;";
+          + "       FORMAT = 'parquet') AS rows \n";
+
+  private static final String createSnapshotTableByRowIdTemplate =
+      createSnapshotTableTemplate + "WHERE rows.datarepo_row_id IN (<datarepoRowIds>);";
```

Comment:
Actually, you might be able to do this with a prepared statement so that we don't need to do the potentially dangerous and less performant thing on line 366.

```suggestion
      createSnapshotTableTemplate + "WHERE rows.datarepo_row_id IN (:datarepoRowIds:);";      
```

I think will work if you just pass in the list of UUIDs (you may need to turn them to strings first)

Reply:
Whoops, I meant:
```createSnapshotTableTemplate + "WHERE rows.datarepo_row_id IN (:datarepoRowIds);";

Reply:
Ok - with the latest changes I'm building the query with both string template and NamedParameterJdbcTemplate. I had to set up a second bean with the configuration to connect to synapse.